model-optimizer/extensions/middle/AddMeanScaleValues.py

```diff
@@ -52,6 +52,16 @@ def insert_pre_processing(graph: Graph, input_node: Node, node_mean_scale_values
         name = input_node.soft_get('name', input_node.id) + '/' + preprocessing_name
         preprocessing = create_op_with_const_inputs(graph, op=op, port_value_dict={1: value}, op_attrs={'name': name})
 
+        if input_node.has_port('out', 0) and len(input_node.out_port(0).get_destinations()) == 1:
+            # There are models with pattern Parameter(uint8) -> Convert(float).
+            # Adding mean/scale leads to the following:
+            # Parameter(uint8) -> Mean/Scale -> Convert(float) which is incorrect.
+            # To fix this mean and scale preprocessing node is inserted after Convert(float) node.
+            out_node = input_node.out_port(0).get_destination().node
+            if out_node.soft_get('type') == "Convert" and (out_node.soft_get('dst_type') in [np.float32, np.float16]):
+                if len(input_node.out_port(0).get_destinations()) == 1:
+                    input_node = out_node
+
```

Comment:
also I worry about value type of preprocessing. Will it be of valid type the same as `dst_type` and is casting correct?

Reply:
We have [override_data_type_of_constant](https://github.com/openvinotoolkit/openvino/blob/fb103e1dd0911ef5c35a4288831585958eab66fd/model-optimizer/extensions/ops/elementwise.py#L16) method in type infer of elementwise operations which aligns const type, so the type will match.

Reply:
Personally, I would not rely on that functionality and make this casting right here. I propose to do it here.
I would like override_data_type_of_constant to report less warnings

Reply:
Done.